src/rpc/blockchain.cpp

```diff
@@ -2646,10 +2649,53 @@ static RPCHelpMan dumptxoutset()
     }
 
     NodeContext& node = EnsureAnyNodeContext(request.context);
+    CConnman& connman = EnsureConnman(node);
+
+    const CBlockIndex* invalidate_index{nullptr};
+    if (!request.params[1].isNull()) {
+        const CBlockIndex* target_index = ParseHashOrHeight(request.params[1], *node.chainman);
+
+        // If the node is running in pruned mode we ensure all necessary block
+        // data is available before starting to roll back.
+        if (node.chainman->m_blockman.IsPruneMode()) {
+            LOCK(::cs_main);
+            const CBlockIndex* current_tip{node.chainman->ActiveChain().Tip()};
+            const CBlockIndex* first_block{node.chainman->m_blockman.GetFirstStoredBlock(*current_tip)};
```

Comment:
This actually isn't a sufficient check because we could be missing the undo data. I am addressing this here: #29668

Reply:
This is fixed now since #29668 got merged and I am using this here now.